lib/ace/mode/sqlserver.js

```diff
@@ -46,6 +46,33 @@ oop.inherits(Mode, TextMode);
 (function() {
     this.lineCommentStart = "--";
     this.blockComment = {start: "/*", end: "*/"};
+    
+    /**
+     * Override keyword completions to ensure proper case for completions
+     * and use smart meta tags instead of 'keyword' for all completions. 
+     */
+    this.getCompletions = function(state, session, pos, prefix) {
+        var keywords = this.$keywordList || this.$createKeywordList();
+        var types = session.$mode.$highlightRules.keywordTypes;
+    
+        return keywords.map(function(word) {
```

Comment:
could we just use types skipping keywordList entirely?
keywordList removes some information that highlighter doesn't need and here we are restoring that using slow indexOf in a large array